python/GafferSceneUI/_SceneViewInspector.py

```diff
@@ -779,6 +779,12 @@ def __edit( self ) :
 		self.__editWindow = _EditWindow( plugs, warning = "\n".join( warnings ) )
 		self.__editWindow.resizeToFitChild()
 		self.__editWindow.popup( self.bound().center() + imath.V2i( 0, 45 ) )
+		self.__editWindow.visibilityChangedSignal().connect( Gaffer.WeakMethod( self.__editWindowVisibilityChanged ), scoped = False )
+
+	def __editWindowVisibilityChanged( self, widget ) :
+
+		if not widget.visible() :
+			self.__editWindow = None
```

Comment:
Is this a necessary part of the fix? This PR seems to have introduced a crash, and my guess is that this may be the reason. To reproduce, edit a string or float value via the inspector, and hit enter with focus in the edit field. This gives me stack traces like the following :

<details>
<summary>FloatPlug Stack</summary>

```
#0  0x00007fffe1c8ff29 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#1  0x00007fffe1c98293 in QApplication::notify(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#2  0x00007fffdd25ccf6 in QApplicationWrapper::notify(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/python2.7/site-packages/PySide2/QtWidgets.so
#3  0x00007fffe2a59c88 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Core.so.5
#4  0x00007fffe1cea6a3 in ?? () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#5  0x00007fffe1c8ff2c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#6  0x00007fffe1c97044 in QApplication::notify(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#7  0x00007fffdd25ccf6 in QApplicationWrapper::notify(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/python2.7/site-packages/PySide2/QtWidgets.so
#8  0x00007fffe2a59c88 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Core.so.5
#9  0x00007fffe22e1a3b in QGuiApplicationPrivate::processKeyEvent(QWindowSystemInterfacePrivate::KeyEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Gui.so.5
```

</details>

<details>
<summary>StringPlug Stack</summary>

```
#1  0x00007fffe2a84873 in QObject::property(char const*) const () from /disk1/john/dev/build/gafferPython2/lib/libQt5Core.so.5
#2  0x00007fffddc0e9c7 in PySide::getWrapperForQObject(QObject*, SbkObjectType*) () from /disk1/john/dev/build/gafferPython2/lib/python2.7/site-packages/PySide2/libpyside2-python2.7.so.5.12
#3  0x00007fffdc5877f3 in QObjectWrapper::eventFilter(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/python2.7/site-packages/PySide2/QtCore.so
#4  0x00007fffe2a59a25 in QCoreApplicationPrivate::sendThroughObjectEventFilters(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Core.so.5
#5  0x00007fffe1c8ff05 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#6  0x00007fffe1c98293 in QApplication::notify(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#7  0x00007fffdd25ccf6 in QApplicationWrapper::notify(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/python2.7/site-packages/PySide2/QtWidgets.so
#8  0x00007fffe2a59c88 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Core.so.5
#9  0x00007fffe1cea6a3 in ?? () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#10 0x00007fffe1c8ff2c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#11 0x00007fffe1c97044 in QApplication::notify(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Widgets.so.5
#12 0x00007fffdd25ccf6 in QApplicationWrapper::notify(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/python2.7/site-packages/PySide2/QtWidgets.so
#13 0x00007fffe2a59c88 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Core.so.5
#14 0x00007fffe22e1a3b in QGuiApplicationPrivate::processKeyEvent(QWindowSystemInterfacePrivate::KeyEvent*) () from /disk1/john/dev/build/gafferPython2/lib/libQt5Gui.so.5
```

</details>

If this is essential for some reason, perhaps we need to resort to `WidgetAlgo.keepUntilIdle()`?

Reply:
Bah. Thanks. I had seen a crash, then failed to recreate it. It seemed wasteful keeping the widget around after the popup was dismissed, but for simplicity, [reverted](https://github.com/GafferHQ/gaffer/pull/3974/commits/42ed8f1f6aba0ee214b9f6a45ef7d9e98561ef91) this bit.